src/pages/blog/streamlit-deployment-https-social-auth.md

```diff
@@ -0,0 +1,122 @@
+
+# Defines how to start the application. In this case the Streamlit application’s filename is st.py, so feel free to adjust it to your app’s name.
+ENTRYPOINT ["streamlit", "run", "st.py", "--server.port=8501", "--server.address=0.0.0.0"]
+```
+
+- **Registry:** Gimlet Registry. This is made by Gimlet for you by default.
+- **Port:** `8501`. This is the default exposed port for Streamlit apps.
+
+When all the setting changes are made like listed above, you can click the **Deploy** button. After that, logs will appear and your app will be set up soon. When deployment status turns `Running`, you can check out the app in your browser by clicking on the link next to the status. Click **Write configuration to Git** to save the deployed application and to be able to edit its settings later.
+
+## Fancy Things for your Streamlit App
+
+You can configure additional settings for your deployed Streamlit app, such as HTTPS certificates and social authentication.
+
+### HTTPS Certificate
+
+You can add HTTPS certificates to your application with Gimlet.
+
+First, you need to configure `cert-manager`, which is an open-source certification issuer.
+
+To do that, click **Environments** in the menu on top and click on your environment’s card. When you first see the environment settings, you have to create gitops repositories, which you can do with a single click.
+
+When the repositories are ready, click **Ingress** in the options on the left, and look for **cert-manager** settings. When you find it, just use the toggle to enable it, and enter an email address where you’d like to be notified when the certificate expires.
+
+You can enable HTTPS with a toggle when you deploy a new application after this.
+
+If you’d like to add an HTTPS certificate to an already deployed app, click the **Environments** button again, then the **Repositories** button next to it. Select the repository just like you did earlier, but this time click the (...) or meatballs menu and click **Edit**.
+
+In the deployment settings, select the **Domain** option, and use the **HTTPS** toggle to enable the certificate, then click **Save**.
+
+### Social Authentication
+
+With Gimlet, you can configure social authentication for GItHub organizations and users. Streamlit Community Cloud has a similar mechanism that allows you to restrict usage to users with the email addresses specified by you.
+
+You can add social authentication pretty much similar to the way you can add a HTTPS certificate.
+
+To be able to add social authentication, you need to do two things:
+- Create a GitHub OAuth application,
+- Configure OAuth2Proxy in Gimlet.
+
+#### Create a GitHub OAuth Application
+
+It’s not necessary, but useful to have the application deployed already so you have the URL generated when you create the application in GitHub.
+
+It’s easiest if you do both at the same time, since some data about the GitHub OAuth application will be needed. Namely these:
```

Comment:
I am confused.

What is both?

Also, don't explain what comes first and why. Just list the steps to so people can follow.

Don't explain why this is the order, just take them from 0 to deployment.